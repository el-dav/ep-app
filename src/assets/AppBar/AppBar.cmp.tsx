import React, { FC } from 'react';
import { PageHeader } from 'antd';

import { styled } from 'theme';

type StyledHeaderProps = {
  appBarType: AppBarType;
};

const ExtraText = styled.h4<StyledHeaderProps>`
  margin: 0px;
  color: ${({ theme }) => theme.palette.primary.contrastText};
  text-anchor: middle;
`;

export enum AppBarType {
  PRIMARY = 'primary',
  SECONDARY = 'secondary'
}

const StyledHeader = styled(PageHeader)<StyledHeaderProps>`
  background-color: ${({ theme, appBarType }) =>
    theme.palette[appBarType].main};
  .ant-page-header-title-view-title {
    color: ${({ theme }) => theme.palette.primary.contrastText};
  }
  .ant-page-header-title-view-extra {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    padding: 0px;
    height: 100%;
    top: 0px;
  }
  width: 100%;
  height: 40px;
  display: flex;
  align-items: center;
`;

type Props = {
  className?: string;
  title: string;
  appBarType?: AppBarType;
  extraText?: string;
};

const AppBar: FC<Props> = ({ className, title, appBarType, extraText }) => (
  <StyledHeader
    className={`${className || ''}`}
    title={title}
    appBarType={appBarType || AppBarType.PRIMARY}
    extra={
      <ExtraText appBarType={appBarType || AppBarType.PRIMARY}>
        {extraText}
      </ExtraText>
    }
  />
);

export default AppBar;
