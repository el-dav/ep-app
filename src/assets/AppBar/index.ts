export { default as AppBar, AppBarType } from './AppBar.cmp';
