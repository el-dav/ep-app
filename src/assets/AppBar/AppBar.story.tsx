import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select } from '@storybook/addon-knobs';

import { ThemeProvider, styled } from 'theme';

import AppBar, { AppBarType } from './AppBar.cmp';

const Container = styled.div`
  width: 100%;
  height: 100%;
`;

storiesOf('AppBar', module).add('Default', () => {
  const title = text('Title', 'This is the title');
  const extraText = text('Extra Text', 'Extra Text');
  const appBarType = select(
    'Trade Type',
    {
      [AppBarType.PRIMARY]: AppBarType.PRIMARY,
      [AppBarType.SECONDARY]: AppBarType.SECONDARY
    },
    AppBarType.PRIMARY
  );
  return (
    <ThemeProvider>
      <Container>
        <AppBar title={title} appBarType={appBarType} extraText={extraText} />
      </Container>
    </ThemeProvider>
  );
});
