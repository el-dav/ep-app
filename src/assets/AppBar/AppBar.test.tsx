import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from 'theme';

import AppBar, { AppBarType } from './AppBar.cmp';

describe('AppBar', () => {
  it('renders the Component correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <AppBar title="Title" />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly with a different AppBarType', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <AppBar title="Title" appBarType={AppBarType.SECONDARY} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly with extra text', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <AppBar title="Title" extraText="Extra text" />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
