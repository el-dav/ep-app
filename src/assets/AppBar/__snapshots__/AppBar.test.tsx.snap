// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders the Component correctly 1`] = `
<div
  className="ant-page-header  css-199n3lt"
>
  <div
    className="ant-page-header-title-view"
  >
    <span
      className="ant-page-header-title-view-title"
    >
      Title
    </span>
    <span
      className="ant-page-header-title-view-extra"
    >
      <h4
        className="css-1u98n6u"
      />
    </span>
  </div>
</div>
`;

exports[`AppBar renders the Component correctly with a different AppBarType 1`] = `
<div
  className="ant-page-header  css-1l6stgg"
>
  <div
    className="ant-page-header-title-view"
  >
    <span
      className="ant-page-header-title-view-title"
    >
      Title
    </span>
    <span
      className="ant-page-header-title-view-extra"
    >
      <h4
        className="css-1u98n6u"
      />
    </span>
  </div>
</div>
`;

exports[`AppBar renders the Component correctly with extra text 1`] = `
<div
  className="ant-page-header  css-199n3lt"
>
  <div
    className="ant-page-header-title-view"
  >
    <span
      className="ant-page-header-title-view-title"
    >
      Title
    </span>
    <span
      className="ant-page-header-title-view-extra"
    >
      <h4
        className="css-1u98n6u"
      >
        Extra text
      </h4>
    </span>
  </div>
</div>
`;
